Add: Morphs Anonce

Honker9999999 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
При шести морфах делает анонс "Внимание! Зафиксированы множественные биоугрозы 6 уровня на [station_name()]. Необходимо уничтожение для продолжения безопасной работы."
Добавлены соотвествующий прок и две переменных (сделан ли анонс и кол-во живых морфов, хотя и второе сделано немного тухедски)
Предложка - https://discord.com/channels/617003227182792704/755125334097133628/1007382716511158352
<!-- Include a small to medium description of what your PR changes. Document every changes or this may delay review or even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Предложка
Ладно
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
## Changelog
:cl:
add: Анонс при шести морфах
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Honker9999999 committed:
Add: Morphs Anonce

larentoun reviewed (commented):
А ты вообще проверил, что анонс будет?

Robustin4 commented:
может переменные глобальными сделать?

Honker9999999 commented:
> может переменные глобальными сделать?

Хз нужно ли, ибо они только в этом файле упоминаются.

Honker9999999 commented:
> А ты вообще проверил, что анонс будет?

Я тестил все разы как менял этот код кроме последнего, ибо там изменение было мелким, но эыые и оно все сломало. Да, действительно теперь не работает, щас починю.

Honker9999999 committed:
Хотфикс

Honker9999999 committed:
Хотфикс-хотфикса

Просто заменил проверку переменная == FALSE на !переменная

Honker9999999 commented:
Жестко нахотфиксил целый ПР хотфиксов

larentoun commented on code/game/gamemodes/miniantags/morph/morph.dm:
```diff
@@ -1,7 +1,7 @@
+#define MORPHS_ANNOUNCE_THRESHOLD 5
 
-var/morphs_number = 0
 var/anonced = FALSE
```
announced

larentoun commented:
В остальном, вроде нормально. Немного грамматики поменять

larentoun commented on code/game/gamemodes/miniantags/morph/morph.dm:
```diff
@@ -62,6 +65,13 @@
 
+proc/check_morphs()
+	if((GLOB.MORPHS_NUMBER > MORPHS_ANNOUNCE_THRESHOLD) && (!anonced))
+		GLOB.command_announcement.Announce("Внимание! Зафиксировано критическое биожизни морфического типа на [station_name()]. Необходимо уничтожить их для восстановления безопасной среды.", "Central Command Biological Affairs", 'sound/AI/commandreport.ogg')
```
"Внимание! Зафиксировано критическое количество биожизни морфического типа на [station_name()]. Необходимо уничтожить их для восстановления безопасной среды."

Honker9999999 committed:
Исправил ошибку в тексте

Да.

Bizzonium commented on code/game/gamemodes/miniantags/morph/morph.dm:
```diff
@@ -62,6 +65,13 @@
 
+proc/check_morphs()
+	if((GLOB.MORPHS_NUMBER > MORPHS_ANNOUNCE_THRESHOLD) && (!announced))
+		GLOB.command_announcement.Announce("Внимание! Зафиксированы множественные био-угрозы 6-М уровня на [station_name()]. Необходимо уничтожение для продолжения безопасной работы.", "Central Command Biological Affairs", 'sound/AI/commandreport.ogg')
```
`био-угрозы 6-М уровня` ?

larentoun replied:
> `био-угрозы 6-М уровня` ?

Видимо, где-то в чатах сказали, что это угроза 6-М. И такой вроде не было у нас ранее

Honker9999999 replied:
Самое логичное что я смог придумать, т.к. он страшнее вируса, но не такой страшный как блоб или терроры. А так ингейм не упоминается какой именно у морфа уровень биогурозы и есть ли он у него.

larentoun replied:
> Самое логичное что я смог придумать, т.к. он страшнее вируса, но не такой страшный как блоб или терроры. А так ингейм не упоминается какой именно у морфа уровень биогурозы и есть ли он у него.

ЭЭЭЭ, НЕТ! Спрашивай у викиделов/лороделов/админов, отсебятины, особенно такой важной, нельзя

Honker9999999 replied:
> > Самое логичное что я смог придумать, т.к. он страшнее вируса, но не такой страшный как блоб или терроры. А так ингейм не упоминается какой именно у морфа уровень биогурозы и есть ли он у него.
> 
> ЭЭЭЭ, НЕТ! Спрашивай у викиделов/лороделов/админов, отсебятины, особенно такой важной, нельзя

Я и есть викидел, так что разрешаю себе, но вообще - спросил у бота, он сказал сначала в игру добавить, потом на вики.

larentoun replied:
Тогда ладно

Honker9999999 committed:
Исправил уровень биоугрозы

По просьбе хелла изменил с 6-М на просто 6-ю биоугрозу. Плюс исправил "Био-угрозы" на "Биоугрозы"

LiquidPotroh commented:
С русским языком, конечно, тут проблемки. Неужели никому не режет слух "критическое биожизни"? Критическое что? Может критическое количество "биожизни"? Биожизнь это вообще не слово, а какой то бред. Необходимо уничтожить "их". Чтобы использовать "их" до этого должны быть какие-нибудь "они", а у автора только "критическое биожизни".

Honker9999999 commented:
> С русским языком, конечно, тут проблемки. Неужели никому не режет слух "критическое биожизни"? Критическое что? Может критическое количество "биожизни"? Биожизнь это вообще не слово, а какой то бред. Необходимо уничтожить "их". Чтобы использовать "их" до этого должны быть какие-нибудь "они", а у автора только "критическое биожизни".

Поэтому собсна исправили. Я забыл это изначально исправить, так что пришлось делать хотфикс.
